core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -722,15 +719,11 @@ public int getBlockletSizeInMB() {
   public String getBucketHashMethod() {
     String configuredMethod = tableInfo.getFactTable().getTableProperties()
         .get(CarbonCommonConstants.BUCKET_HASH_METHOD);
-    if (configuredMethod == null) {
-      return CarbonCommonConstants.BUCKET_HASH_METHOD_DEFAULT;
-    } else {
-      if (CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE.equals(configuredMethod)) {
-        return CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE;
-      }
-      // by default we use spark_hash_expression hash method
-      return CarbonCommonConstants.BUCKET_HASH_METHOD_DEFAULT;
+    if (CarbonCommonConstants.BUCKET_HASH_METHOD_NATIVE.equals(configuredMethod)) {
```

Comment:
should keep (not) null check

Reply:
"==" and "instanceof" contain null check

Reply:
ok